### DISCUSSION TITLE
Can't connect to Milvus with Attu

### DISCUSSION THREAD

**Discussion by jmugan** [milvus-io/milvus#16559] [Q&A and General discussion] [answered]
From the cli I can do this, so it is running
```
milvus_cli > show connection
+-------+-----------+
| Host  | 127.0.0.1 |
| Port  |   19530   |
| Alias |  default  |
+-------+-----------+
```
When I try Attu with 127.0.0.1:19530 I get the error "BadRequestError: Connect milvus failed, check your milvus address."

**Comment by shanghaikid** [comment, marked as answer]
Where is your attu installed?  if you try to connect to 127.0.0.1, attu must be installed in the same network.

**Reply by nameczz** [reply]
if run Attu by docker, i don't think 127.0.0.1:19530 is allowed.  Need use ip instead, like : 192.168.1.108:19530

**Reply by jmugan** [reply]
Yeah, it's docker. The instructions show running it with docker and also show 127.0.0.1:19530, but they are probably two incompatible pieces stuck together. I'll try the ip, thanks!

**Reply by jmugan** [reply]
Using my ip address didn't work either. Same response that I should check my milvus address. Weird.

**Reply by nameczz** [reply]
If milvus ip  and attu ip are both 192.168.1.108, docker command should looks like:
docker run -p 8000:3000 -e HOST_URL=http://192.168.1.108:8000 -e MILVUS_URL= 192.168.1.108:19530 zilliz/attu:latest

If not same, please check these network are interconnected.

Also if you are using vpn  ,also will cause this problem.

**Reply by jmugan** [reply]
Yes, that's what I have. What do you mean checking that the network are connected?

**Reply by jmugan** [reply]
With that, it keeps saying "Network error" with no other information.

**Reply by jmugan** [reply]
I can do http://0.0.0.0:8000/api/v1/healthy but I can't change 0.0.0.0 to my ip address.

**Reply by nameczz** [reply]
Network error means HOST_URL not right, browser send request to Attu server(HOST_URL), but server has no response.

**Reply by nameczz** [reply]
http://0.0.0.0:8000/api/v1/healthy is working, but http://{ip}:8000/api/v1/healthy is not working?

Could you paste your docker command? And you can use fake ip instead of real ip .

**Reply by jmugan** [reply]
Right. It is `docker run -p 8000:3000 -e HOST_URL=http://72.999.999.999:8000 -e MILVUS_URL=72.999.999.999:19530 zilliz/attu:latest`

**Reply by shanghaikid** [reply]
weird. if you can access 0.0.0.0:8000, that means the docker service is working.  
Is your IP correct?

**Reply by nameczz** [reply]
Weird, docker command looks good to me.
so if use docker run -p 8000:3000 -e HOST_URL=http://0.0.0.0:8000 -e MILVUS_URL=0.0.0.0:19530 zilliz/attu:latest is working for you ?

**Reply by jmugan** [reply]
As far as I can tell. It's the IP that comes up when I do `curl ifconfig.me`.

**Reply by jmugan** [reply]
Running `docker run -p 8000:3000 -e HOST_URL=http://0.0.0.0:8000/ -e MILVUS_URL=0.0.0.0:19530 zilliz/attu:latest` gives me `"BadRequestError: Connect milvus failed, check your milvus address."`

**Reply by nameczz** [reply]
Where does MIlvus Cli run? Same machine with attu ? Could you please try to use Milvus cli connect Milvus with {ip}:19530. Not 127.0.0.1:19530.

**Reply by shanghaikid** [reply]
> 

I see, please use your local IP, not the wan IP.    

Instead, you can get your IP using ```ipconfig``` on windows  or ```ifconfig``` if on mac or linux.

**Reply by jmugan** [reply]
Yep, that was it! Thanks @shanghaikid and @nameczz! When I was getting my IP it was getting me the IP from the point of view of the outside world (the wan IP) but I needed the inside IP (I'm just running everything on my laptop). Which I got from wifi settings.

**Comment by echo-initial** [comment]
my milvus standalone and Attu are both installed on the cloud service. I can access the login page, but can't log in

run  ```docker run -p 8000:3000  -e MILVUS_URL={Server IP}:19530 zilliz/attu:latest```

log information
```
Error: Error: 14 UNAVAILABLE: No connection established
    at /app/node_modules/@zilliz/milvus2-sdk-node/dist/utils/index.js:28:15
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  expose: true,
  statusCode: 400,
  status: 400
}
POST /api/v1/milvus/connect 400 Error: Error: 14 UNAVAILABLE: No connection established

 POST undefined /api/v1/milvus/connect undefined ms @ Mon, 06 Mar 2023 09:52:05 GMT ::ffff:101.207.139.251 from undefined Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.92 Safari/537.36
```


However, I can successfully log in to milvus standalone, which is installed on the Windows machine with the attu connection cloud service

**Reply by shanghaikid** [reply]
Please make sure that the attu docker can access the milvus service.

**Reply by jmugan** [reply]
Hi, how do you check that attu docker can access the milvus service? I'm now getting the same error. Both milvus and attu are running in docker.

**Reply by shanghaikid** [reply]
> Hi, how do you check that attu docker can access the milvus service? I'm now getting the same error. Both milvus and attu are running in docker.

1. first `docker ps` to get your container ID
2. then `docker inspect your-container-ID -f "{{json .NetworkSettings.Networks }}"`
3. make sure those tow IPAddresses can access each other.

**Reply by jmugan** [reply]
Thanks. I did the check you suggested and since I started Attu on the milvus network, I see that they are on the same network. But I still get that error. Weird.

**Reply by shanghaikid** [reply]
> Thanks. I did the check you suggested and since I started Attu on the milvus network, I see that they are on the same network. But I still get that error. Weird.

Please make sure the address that you entered in the attu's address input can be accessed from attu's docker.

**Reply by jmugan** [reply]
I got it to stop saying the network error, I think, but now looking in the logs attu says
`POST /api/v1/milvus/connect 400 Error: Error: 12 UNIMPLEMENTED: unknown method CheckHealth for service milvus.proto.milvus.MilvusService`

**Reply by shanghaikid** [reply]
> I got it to stop saying the network error, I think, but now looking in the logs attu says `POST /api/v1/milvus/connect 400 Error: Error: 12 UNIMPLEMENTED: unknown method CheckHealth for service milvus.proto.milvus.MilvusService`

what's the version of your milvus? I think if you got this error, your milvus is a little bit old.
